### PR TITLE
docstore: introduce OrderBy method

### PR DESCRIPTION
**Pull request by jba**
- Add an OrderBy method to docstore.Query.

- Add fields in driver.Query.

- Overhaul and improve query error checking.

OrderBy implementations TBD.

### CODE REVIEW

File: internal/docstore/query.go [hunk 1]
```diff
@@ -112,58 +156,65 @@ func (q *Query) BeforeQuery(f func(asFunc func(interface{}) bool) error) *Query
...
 	wrapErr := func(err error) error { return wrapError(dcoll, err) }
-	if err := q.init(fps); err != nil {
+	if err := q.initGet(fps); err != nil {
 		return &DocumentIterator{err: wrapErr(err)}
 	}
 	it, err := dcoll.RunGetQuery(ctx, q.dq)
 	return &DocumentIterator{iter: it, wrapError: wrapErr, err: wrapErr(err)}
 }
 
+func (q *Query) initGet(fps []FieldPath) error {
```
Comment (shantuo) [inline internal/docstore/query.go]: Should this also check whether `OrderByField` also appears in the filters?

Comment (jba) [reply internal/docstore/query.go]: Done.

I also added implementations for mem, mongo and firestore.

PTAL.

Comment (shantuo) [reply internal/docstore/query.go]: Forgot to push?

Comment (jba) [reply internal/docstore/query.go]: Right :(

PTAL

**Pull request merged**